### ISSUE TITLE
Regression since 0.10.7: Unable to unmarshall response (Data read has a different checksum than expected.)

### ISSUE THREAD

**Issue by stuartleylandcole** [localstack/localstack#1978]
I have just pulled the `latest` Docker image and I'm getting

```
Unable to unmarshall response (Data read has a different checksum than expected.)
```

errors when using `S3Client#putObject` from the AWS SDK. This is almost identical to #1611 from October.

Fixing to version 0.10.7 fixes the issue. Looking at the commits since this version was released (09-Jan-2020) the following PRs are the only ones that mention S3:

- https://github.com/localstack/localstack/pull/1965
- https://github.com/localstack/localstack/pull/1976

**Comment by whummer** [comment]
Thanks for reporting @stuartleylandcole . Can you please provide a small reproducible example. The previous issue in #1611 (seems to have been related to multi-part uploads) has been fixed and should be covered with this test case: https://github.com/localstack/localstack/blob/master/tests/integration/test_s3.py#L328-L355 . Please provide some more details to help reproduce this particular issue. Thanks

**Comment by stuartleylandcole** [comment]
Hi @whummer. I'm not sure my specific case from #1611 was because of multi-part uploads as I was storing an empty body. Nevertheless this code will replicate the issue:

```
final S3Client s3Client = S3Client.builder()
                .httpClient(UrlConnectionHttpClient.builder().build())
                .credentialsProvider(DefaultCredentialsProvider.create())
                .region(Region.EU_WEST_1)
                .endpointOverride(URI.create("http://localhost:4572"));

s3Client.putObject(PutObjectRequest.builder()
                        .bucket(fullBucketName)
                        .key(contentDetailsId)
                        .build(),
                RequestBody.empty());
```

Same behaviour on versions 2.10.56 and 2.9.7 of the AWS SDK v2 for Java.

Possibly of use is this additional logging from the latest version of the SDK:

```
software.amazon.awssdk.core.exception.SdkClientException: Data read has a different checksum than expected. Was 0xd41d8cd98f00b204e9800998ecf8427e, but expected 0xec355f955798b44c8f00022b1d09af56
```

(Previously the log just stated `Data read has a different checksum than expected`)

Hopefully this is useful, please let me know if you need anything else!

**Comment by whummer** [comment]
Thanks for providing the details @stuartleylandcole . This should be fixed in #1993 - can you please give it a try with the `latest` version? Please report here if the problem persists. Thanks

**Comment by stuartleylandcole** [comment]
@whummer just tested and all working fine, thanks for the quick fix!
